### PR TITLE
ECO-2162 Refactor templates

### PR DESCRIPTION
**Pull request by FRITS-Kh**
Ticket: https://spryker.atlassian.net/browse/ECO-2162

### CODE REVIEW

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+declare var csf: any;
+
+import Component from 'ShopUi/models/component';
+import ScriptLoader from 'ShopUi/components/molecules/script-loader/script-loader';
+
+// Define a custom style.
+const styleObject = {
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: `stylesConfig`?

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                        hostedCardNumberField : '4111 1111 1111 1111',
+                        hostedExpiryDateField : '08/18',
+                        hostedSecurityCodeField : '737'
+                    }
+                }
+            }
+        }
+    }
+
+    get stylesIFramesConfig(): any {
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: `stylesIframesConfig`

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    'client-config-key': data.configKey
+} %}
+
+{% block body %}
+    {% include molecule('script-loader') with {
+        attributes: {
+            src: data.scriptSrc
+        }
+    } only %}
+    <div class="col col--sm-12 col--md-6 js-chckt-pm__pm-holder {{ config.jsName }}__form">
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig]: `js-chckt-pm__pm-holder` is this class needed?

File: src/SprykerEco/Yves/Adyen/Theme/default/views/credit-card/credit-card.twig [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/credit-card/credit-card.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/ideal/ideal.twig [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/ideal/ideal.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/klarna-invoice/klarna-invoice.twig [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/klarna-invoice/klarna-invoice.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/paypal/paypal.twig [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/paypal/paypal.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/prepayment/prepayment.twig [hunk 8]
```diff
@@ -0,0 +1,12 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/prepayment/prepayment.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/sofort/sofort.twig [hunk 9]
```diff
@@ -0,0 +1,12 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/sofort/sofort.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/wechatpay/wechatpay.twig [hunk 10]
```diff
@@ -0,0 +1,12 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/wechatpay/wechatpay.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/klarna-invoice/klarna-invoice.twig [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+} %}
+
+{% block body %}
+    <div class="col col--sm-12 col--md-6">
+        {{ form_widget(data.form.adyenKlarnaInvoice) }}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/klarna-invoice/klarna-invoice.twig]: But do we really need templates for it? It's just regular form field rendering.

File: src/SprykerEco/Yves/Adyen/Theme/default/views/ideal/ideal.twig [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+} %}
+
+{% block body %}
+    <div class="col col--sm-12 col--md-6">
+        <div class="form__fields">
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/ideal/ideal.twig]: Do we need additional div containers here?

File: src/SprykerEco/Yves/Adyen/Theme/default/views/alipay/alipay.twig [hunk 13]
```diff
@@ -0,0 +1,12 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/views/alipay/alipay.twig]: Please remove line-break.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    protected mapEvents(): void {
+        this.scriptLoader.addEventListener('scriptload', (event: Event) => this.onScriptLoad(event));
+    }
+
+    protected onScriptLoad(event: Event): void {
+        this.initIframes();
+    }
+
+    protected initIframes(): void {
+        // Load Secured Fields.
```
[the remark this answers is not in the document]
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: Fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+        const securedFields = csf(this.hostedIFramesConfig());
+
+        // Triggers when receiving a brand callback from the credit card number validation.
```
[the remark this answers is not in the document]
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: Fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+        securedFields.onBrand(function (brandObject) {
+            if (brandObject.brand) {
+                document.getElementById('card-type').innerHTML = brandObject.brand;
+            }
+        });
+    }
+
+    protected hostedIFramesConfig(): any {
```
[the remark this answers is not in the document]
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: Fixed

File: src/SprykerEco/Yves/Adyen/Form/DirectDebitSubForm.php [hunk 17]
```diff
@@ -17,7 +17,7 @@
 
 class DirectDebitSubForm extends AbstractSubForm
 {
-    protected const PAYMENT_METHOD = 'direct_debit';
+    protected const PAYMENT_METHOD = 'direct-debit';
```
[the remark this answers is not in the document]
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Adyen/Form/DirectDebitSubForm.php]: It's ok.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts (hunk 1 above)
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: Changed

File: src/SprykerEco/Yves/Adyen/Form/CreditCardSubForm.php [hunk 18]
```diff
@@ -19,7 +19,7 @@ class CreditCardSubForm extends AbstractSubForm
     public const SDK_CHECKOUT_SECURED_FIELDS_URL = 'sdkUrl';
     public const SDK_CHECKOUT_ORIGIN_KEY = 'sdkOriginKey';
 
-    protected const PAYMENT_METHOD = 'credit_card';
+    protected const PAYMENT_METHOD = 'credit-card';
```
[the remark this answers is not in the document]
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Adyen/Form/CreditCardSubForm.php]: It's ok.

File: src/SprykerEco/Yves/Adyen/Form/KlarnaInvoiceSubForm.php [hunk 19]
```diff
@@ -18,7 +18,7 @@
 
 class KlarnaInvoiceSubForm extends AbstractSubForm
 {
-    protected const PAYMENT_METHOD = 'klarna_invoice';
+    protected const PAYMENT_METHOD = 'klarna-invoice';
```
[the remark this answers is not in the document]
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Adyen/Form/KlarnaInvoiceSubForm.php]: It's ok.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts (hunk 2 above)
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig (hunk 3 above)
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig]: removed

File: src/SprykerEco/Yves/Adyen/Theme/default/views/klarna-invoice/klarna-invoice.twig (hunk 11 above)
Comment (FRITS-Kh) [reply src/SprykerEco/Yves/Adyen/Theme/default/views/klarna-invoice/klarna-invoice.twig]: I think the question for @aleksey-kotsuba

**Pull request merged**